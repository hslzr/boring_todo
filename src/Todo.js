import React, { useState } from 'react'

const Todo = (props) => {
  
  const [completed, setCompleted] = useState(props.task.completed)

  const handleClick = () => {
    setCompleted( !completed )
  }

  return (
    <li className="flex content-stretch">
      <div
        className={"br2 ph3 pv2 mb2 " + (completed ? 'bg-near-white' : 'bg-light-gray')}
        style={{flexGrow: 2}}
        onClick={handleClick}>
        <input
          className={"pa2 bn " + (completed ? 'bg-near-white silver strike' : 'bg-light-gray near-black')}
          value={props.task.text}
          //disabled={ !completed }
          //readOnly={completed}
          style={{outline: 0}}
          onChange={() =>{}} />
      </div>
      <div
        className="br2 pa3 pointer mb2 ml2 dim bg-near-white red">
        Del
      </div>
    </li>
  )
}

export default Todo
