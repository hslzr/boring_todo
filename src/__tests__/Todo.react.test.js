import React from 'react'
import renderer from 'react-test-renderer'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Todo from '../Todo'

configure({ adapter: new Adapter() })

test('Todo exists? idk', () => {
  const component = renderer.create(
    <Todo 
      task={{text: 'Demo test', completed: false}}
      index={1234}
      key={1234} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
