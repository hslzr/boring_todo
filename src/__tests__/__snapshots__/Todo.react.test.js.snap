// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo exists? idk 1`] = `
<li
  className="flex content-stretch todo-item"
>
  <div
    className="todo-complete-button bg-green near-white br2 ph3 pv1 pointer mb2 mr2 dim flex items-center"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="20 6 9 17 4 12"
      />
    </svg>
  </div>
  <div
    className="todo-content br2 ph3 pv1 mb2 bg-light-gray"
    style={
      Object {
        "flexGrow": 2,
      }
    }
  >
    <textarea
      className="pa2 bn w-100 bg-light-gray near-black"
      disabled={false}
      onChange={[Function]}
      style={
        Object {
          "outline": 0,
          "resize": "none",
        }
      }
      value="Demo test"
    />
  </div>
  <div
    className="br2 ph3 pv1 pointer mb2 ml2 dim bg-near-white red flex items-center"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height={18}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={18}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="3 6 5 6 21 6"
      />
      <path
        d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
      />
    </svg>
  </div>
</li>
`;
